### ISSUE TITLE
Need more Information of minimumConnectionTimeMs and keepalive Property and relation between them

### ISSUE THREAD

**Issue by harshkoralwala** [aws/aws-iot-device-sdk-js#364]
Hello all,

I need more information on minimumConnectionTimeMs and keepalive Property.

Is there any relation between them?

**Observation** **1**:

keepalive:60 [added in configuration manually]
minimumConnectionTimeMs:20000[not added in configuration,so this will be added with default value while device object created] as per documentation

After some time(near to 10 minutes), I got an error of premature close every time. Don't know why?policy attached and policy has a certificate assigned along with it the policy has all the needed access of IoT i.e publish, connect, subscribe, receive

**Observation** **2**:
keepalive:10 [added in configuration manually]
minimumConnectionTimeMs:20000[not added in configuration,so this will be added with default value while device object created] as per documentation

Here no premature close error raised and functionality works totally fine.

**In summary**, _I can solve the issue by assigning a keepalive value lower than the minimumConnectionTimeMs value._

So I need help to understand the relation between these 2 properties and how they internally work

Thanks in advance.

**Comment by jmklix** [comment]
What is the full error that you get with observation 1? Also what happens when you leave both keepalive and minimumConnectionTimeMs at the default values?

As noted [here](https://docs.amazonaws.cn/en_us/general/latest/gr/iot-core.html) the minimum keepalive is 30 seconds. So that might be causing some unexpected behaviors.
